src/DotNet.Status.Web/TimelineIssueTriageService.cs

```diff
@@ -0,0 +1,286 @@
+                    SortDirection = SortDirection.Ascending,
+                };
+                openIssues.Labels.Add(_labelName);
+
+                _logger.LogInformation("Getting open issues");
+                var existingTriageIssues = await gitHubClient.Issue.GetAllForRepository(issuePayload.Repository.Id, openIssues);
+                _logger.LogInformation($"There are {existingTriageIssues.Count} open issues with the '{_labelName}' label");
+                foreach (var existingIssue in existingTriageIssues)
+                {
+                    if (existingIssue.Number != issuePayload.Issue.Number)
+                    {
+                        var existingIssueItems = GetTriageItemsProperties(existingIssue.Body);
+                        if (IsDuplicate(triageItems, existingIssueItems))
+                        {
+                            await gitHubClient.Issue.Comment.Create(issuePayload.Repository.Id, issuePayload.Issue.Number, $"Duplicate issue was detected.\n\nClosing as duplicate of {existingIssue.HtmlUrl}\n\nFor more information see {_docLink}");
+                            var issueUpdate = new IssueUpdate
+                            {
+                                State = ItemState.Closed,
+                            };
+                            await gitHubClient.Issue.Update(issuePayload.Repository.Id, issuePayload.Issue.Number, issueUpdate);
+
+                            return;
+                        }
+                    }
+                }
+
+                // No duplicates, add label and move issue to triage
+                var issue = await gitHubClient.Issue.Get(issuePayload.Repository.Id, issuePayload.Issue.Number);
+                if (!issue.Labels.Any(l => l.Name == _labelName))
+                {
+                    var update = issue.ToUpdate();
+                    update.AddLabel(_labelName);
+                    await gitHubClient.Issue.Update(issuePayload.Repository.Id, issuePayload.Issue.Number, update);
+
+                    // add into notification epic -> currently 8/2020 its First Response epic
+                    NotificationEpicOptions epic = _githubOptions.Value.NotificationEpic;
+
+                    if (epic != null)
+                    {
+                        var epicRepoData = await gitHubClient.Repository.Get(_githubOptions.Value.Organization, epic.Repository);
+
+                        _logger.LogInformation("Adding the issue to ZenHub Epic...");
+                        await _zenHub.AddIssueToEpicAsync(
+                            new ZenHubClient.IssueIdentifier(issuePayload.Repository.Id, issue.Number),
+                            new ZenHubClient.IssueIdentifier(epicRepoData.Id, epic.IssueNumber)
+                        );
+                    }
+                    else
+                    {
+                        _logger.LogInformation("No ZenHub epic configured, skipping...");
+                    }
+                }
```

Comment:
This could be extracted as some `LabelAndAssignToEpic` method maybe? The parent method is getting a bit long